lib/vagrant-cloudstack/action/run_instance.rb

```diff
@@ -191,129 +177,39 @@ def call(env)
-            end
-          end
+          enable_static_nat(env, networkingConfig)
 
           pf_private_rdp_port = 3389
-          pf_private_rdp_port = env[:machine].config.vm.rdp.port if ( env[:machine].config.vm.respond_to?(:rdp) && env[:machine].config.vm.rdp.respond_to?(:port) )
+          pf_private_rdp_port = env[:machine].config.vm.rdp.port if (env[:machine].config.vm.respond_to?(:rdp) && env[:machine].config.vm.rdp.respond_to?(:port))
 
-          if pf_private_port.nil?
+          networkingConfig.pf_private_rdp_port = pf_private_rdp_port
+
+          vm_guest = env[:machine].config.vm.guest || :linux
+          if networkingConfig.pf_private_port.nil?
             communicator = env[:machine].communicate.instance_variable_get('@logger').instance_variable_get('@name')
             comm_obj = env[:machine].config.send(communicator)
 
-            pf_private_port = comm_obj.port if comm_obj.respond_to?('port')
-            pf_private_port = comm_obj.guest_port if comm_obj.respond_to?('guest_port')
-            pf_private_port = comm_obj.default.port if (comm_obj.respond_to?('default') && comm_obj.default.respond_to?('port'))
-          end
-
-          if (pf_ip_address_id || pf_ip_address) && (pf_public_port || pf_public_port_randomrange)
-            port_forwarding_rule = {
-              :ipaddressid  => pf_ip_address_id,
-              :ipaddress    => pf_ip_address,
-              :protocol     => 'tcp',
-              :publicport   => pf_public_port,
-              :privateport  => pf_private_port,
-              :openfirewall => pf_open_firewall
-            }
-
-            pf_public_port = domain_config.pf_public_port = create_randomport_forwarding_rule(
-              env,
-              port_forwarding_rule,
-              pf_public_port_randomrange[:start]...pf_public_port_randomrange[:end],
-              'pf_public_port'
-            )
-
-            if pf_trusted_networks && !pf_open_firewall
-              # Allow access to public port from trusted networks only
-              fw_rule_trusted_networks = {
-                  :ipaddressid  => pf_ip_address_id,
-                  :ipaddress    => pf_ip_address,
-                  :protocol     => 'tcp',
-                  :startport    => pf_public_port,
-                  :endport      => pf_public_port,
-                  :cidrlist     => pf_trusted_networks.join(',')
-              }
-              firewall_rules = [] unless firewall_rules
-              firewall_rules << fw_rule_trusted_networks
-            end
+            networkingConfig.pf_private_port = comm_obj.port if comm_obj.respond_to?('port')
+            networkingConfig.pf_private_port = comm_obj.guest_port if comm_obj.respond_to?('guest_port')
+            networkingConfig.pf_private_port = comm_obj.default.port if (comm_obj.respond_to?('default') && comm_obj.default.respond_to?('port'))
+            # networkingConfig.pf_private_port ||= vm_guest == :linux ? '22' : '5985'
```

Comment:
@bheuvel found another one. But I'm not sure what to do with this one. Is it needed?